app/forms/image-upload.tsx

```diff
@@ -640,3 +655,145 @@ export function CreateImageSideModalForm() {
+  )
+}
+
+function BootableNotice({
+  efiPartOffset,
+  isBootableCd,
+  isCompressed,
+}: {
+  efiPartOffset: number
+  isBootableCd: boolean
+  isCompressed: boolean
+}) {
+  // this message should only appear if the image doesn't have a header
+  // marker we are looking for and does not appear to be compressed
+  const efiPartOrBootable = efiPartOffset !== -1 || isBootableCd
+  if (efiPartOrBootable && !isCompressed) return null
+
+  const content = (
+    <div className="flex flex-col space-y-2">
+      <ul className="ml-4 list-disc">
+        {!efiPartOrBootable && (
+          <li>
+            <div>Bootable markers not found at any block size.</div>
+            <div>
+              Expected either “EFI PART” marker at offsets 512 / 2048 / 4096 or “CD001” at
+              offset 0x8001 (for a bootable CD).
+            </div>
+          </li>
+        )}
+        {isCompressed && (
+          <li>
+            <div>This might be a compressed image.</div>
+            <div>
+              Only raw, uncompressed images are supported. Files such as qcow2, vmdk,
+              img.gz, iso.7z may not work.
+            </div>
+          </li>
+        )}
+      </ul>
+      <div>
+        Learn more about{' '}
+        <a
+          target="_blank"
+          rel="noreferrer"
+          href={links.preparingImagesDocs}
+          className="inline-flex items-center underline"
+        >
+          preparing images for import
+          <OpenLink12Icon className="ml-1" />
+        </a>
+      </div>
+    </div>
+  )
+
+  return (
+    <Message
+      variant="info"
+      title="This image might not be bootable"
+      className="[&>*]:space-y-2"
+      content={content}
+    />
+  )
+}
+
+async function readAndMatch(
+  file: File,
+  offset: number,
+  length: number,
+  matchString: string
+): Promise<boolean> {
+  const slice = file.slice(offset, offset + length)
+  const reader = new FileReader()
+
+  const promise = new Promise<boolean>((resolve, reject) => {
+    reader.onloadend = (e) => {
+      if (!e || !e.target) {
+        resolve(false)
+        return
+      }
+
+      if (e.target.readyState === FileReader.DONE) {
+        if (e.target.result instanceof ArrayBuffer) {
+          const actualHeader = String.fromCharCode(...new Uint8Array(e.target.result))
+          resolve(actualHeader === matchString)
+        } else {
+          resolve(false)
+        }
+      }
+    }
+
+    reader.onerror = (error) => {
+      console.error(`Error reading file at offset ${offset}:`, error)
+      reject(error)
+    }
+  })
+
+  reader.readAsArrayBuffer(slice)
+  return await promise
+}
+
+async function getEfiPartOffset(file: File) {
+  const offsets = [512, 2048, 4096]
+  for (const offset of offsets) {
+    const isMatch = await readAndMatch(file, offset, 8, 'EFI PART')
+    if (isMatch) return offset
+  }
+  return -1
+}
+
+const compressedExts = ['.gz', '.7z', '.qcow2', '.vmdk']
```

Comment:
I think this all looks good, my main concern was getting the contents of a list like this right. @iliana ?